### ISSUE TITLE
`aws_sdk_dynamodb::client::Builder::new().rustls()` no longer work as of v0.7.0

### ISSUE THREAD

**Issue by nmoutschen** [awslabs/aws-sdk-rust#467]
### What is the problem?

When creating a client using the `Builder` and the `.rustls()` method, the builder no longer works.

### Version

├── aws-config v0.7.0 │   ├── aws-http v0.7.0 │   │   ├── aws-smithy-http v0.37.0 │   │   │   ├── aws-smithy-types v0.37.0 │   │   ├── aws-smithy-types v0.37.0 (*) │   │   ├── aws-types v0.7.0 │   │   │   ├── aws-smithy-async v0.37.0 │   │   │   ├── aws-smithy-client v0.37.0 │   │   │   │   ├── aws-smithy-async v0.37.0 (*) │   │   │   │   ├── aws-smithy-http v0.37.0 (*) │   │   │   │   ├── aws-smithy-http-tower v0.37.0 │   │   │   │   │   ├── aws-smithy-http v0.37.0 (*) │   │   │   │   ├── aws-smithy-types v0.37.0 (*) │   │   │   ├── aws-smithy-types v0.37.0 (*) │   ├── aws-sdk-sso v0.7.0 │   │   ├── aws-endpoint v0.7.0 │   │   │   ├── aws-smithy-http v0.37.0 (*) │   │   │   ├── aws-types v0.7.0 (*) │   │   ├── aws-http v0.7.0 (*) │   │   ├── aws-sig-auth v0.7.0 │   │   │   ├── aws-sigv4 v0.7.0 │   │   │   │   ├── aws-smithy-http v0.37.0 (*) │   │   │   ├── aws-smithy-http v0.37.0 (*) │   │   │   ├── aws-types v0.7.0 (*) │   │   ├── aws-smithy-async v0.37.0 (*) │   │   ├── aws-smithy-client v0.37.0 (*) │   │   ├── aws-smithy-http v0.37.0 (*) │   │   ├── aws-smithy-http-tower v0.37.0 (*) │   │   ├── aws-smithy-json v0.37.0 │   │   │   └── aws-smithy-types v0.37.0 (*) │   │   ├── aws-smithy-types v0.37.0 (*) │   │   ├── aws-types v0.7.0 (*) │   ├── aws-sdk-sts v0.7.0 │   │   ├── aws-endpoint v0.7.0 (*) │   │   ├── aws-http v0.7.0 (*) │   │   ├── aws-sig-auth v0.7.0 (*) │   │   ├── aws-smithy-async v0.37.0 (*) │   │   ├── aws-smithy-client v0.37.0 (*) │   │   ├── aws-smithy-http v0.37.0 (*) │   │   ├── aws-smithy-http-tower v0.37.0 (*) │   │   ├── aws-smithy-query v0.37.0 │   │   │   ├── aws-smithy-types v0.37.0 (*) │   │   ├── aws-smithy-types v0.37.0 (*) │   │   ├── aws-smithy-xml v0.37.0 │   │   ├── aws-types v0.7.0 (*) │   ├── aws-smithy-async v0.37.0 (*) │   ├── aws-smithy-client v0.37.0 (*) │   ├── aws-smithy-http v0.37.0 (*) │   ├── aws-smithy-http-tower v0.37.0 (*) │   ├── aws-smithy-json v0.37.0 (*) │   ├── aws-smithy-types v0.37.0 (*) │   ├── aws-types v0.7.0 (*) ├── aws-sdk-dynamodb v0.7.0 │   ├── aws-endpoint v0.7.0 (*) │   ├── aws-http v0.7.0 (*) │   ├── aws-sig-auth v0.7.0 (*) │   ├── aws-smithy-async v0.37.0 (*) │   ├── aws-smithy-client v0.37.0 (*) │   ├── aws-smithy-http v0.37.0 (*) │   ├── aws-smithy-http-tower v0.37.0 (*) │   ├── aws-smithy-json v0.37.0 (*) │   ├── aws-smithy-types v0.37.0 (*) │   ├── aws-types v0.7.0 (*) ├── aws-smithy-client v0.37.0 (*)

### Platform

Linux CENSORED 5.10.60.1-microsoft-standard-WSL2 #1 SMP Wed Aug 25 23:20:18 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### AWS Services

aws_sdk_dynamodb

### Description

Here is my code that worked with `aws_sdk_dynamodb` version 0.7.0 and `aws_smithy_http` version 0.37.0:

```rust
let inner_client = aws_sdk_dynamodb::client::Builder::new()
    .rustls()
    .middleware(DynMiddleware::new(middleware))
    .build();
let client = aws_sdk_dynamodb::Client::with_config(inner_client, config);
```

I had to change my code to this for it to work:

```rust
let connector = aws_smithy_client::hyper_ext::Adapter::builder().build(aws_smithy_client::conns::https());
let inner_client = aws_sdk_dynamodb::client::Builder::new()
    .connector(DynConnector::new(connector))
    .middleware(DynMiddleware::new(middleware))
    .build();
let client = aws_sdk_dynamodb::Client::with_config(inner_client, config);
```

### Logs

As of version 0.7.0 and 0.37.0, using `.rustls()` returns the following error:

```
  --> src/main.rs:24:56
   |
24 |     let client = aws_sdk_dynamodb::Client::with_config(inner_client, config);
   |                                                        ^^^^^^^^^^^^ expected struct `DynConnector`, found struct `Adapter`
   |
   = note: expected struct `aws_smithy_client::Client<DynConnector, DynMiddleware<DynConnector>>`
              found struct `aws_smithy_client::Client<Adapter<hyper_rustls::connector::HttpsConnector<hyper::client::connect::http::HttpConnector>>, DynMiddleware<_>>`
```

**Comment by Velfi** [comment]
Hey @nmoutschen, I probably broke this during my refactor of connectors. Sorry about that; I'm investigating a fix.

**Comment by jdisanti** [comment]
Hmmm... Looks like I unintentionally rigged this up for auto-close 😅 Reopening until the fix goes out with the next release.

**Comment by jdisanti** [comment]
The fix for this went out in [v0.8.0](https://github.com/awslabs/aws-sdk-rust/releases/tag/v0.8.0).
